src/main.rs

```diff
@@ -5,16 +5,26 @@ mod setup_scripts;
 pub mod tests;
 pub mod utils;
 
```

Comment:
We should use anyhow/eyre and have proper error handling

Reply:
this would be part of another PR, @cchudant already created one, I will rebase that after this PR